extensions/authentication-jwt/README.md

```diff
@@ -0,0 +1,317 @@
+   import {HttpErrors} from '@loopback/rest';
+   import {securityId, UserProfile} from '@loopback/security';
+   import {compare} from 'bcryptjs';
+   // User --> MyUser
+   import {MyUser} from '../models';
+   // UserRepository --> MyUserRepository
+   import {MyUserRepository} from '../repositories';
+
+   export type Credentials = {
+     email: string;
+     password: string;
+   };
+
+   // User --> MyUser
+   export class CustomUserService implements UserService<MyUser, Credentials> {
+     constructor(
+       // UserRepository --> MyUserRepository
+       @repository(MyUserRepository) public userRepository: MyUserRepository,
+     ) {}
+
+     // User --> MyUser
+     async verifyCredentials(credentials: Credentials): Promise<MyUser> {
+       const invalidCredentialsError = 'Invalid email or password.';
+
+       const foundUser = await this.userRepository.findOne({
+         where: {email: credentials.email},
+       });
+       if (!foundUser) {
+         throw new HttpErrors.Unauthorized(invalidCredentialsError);
+       }
+
+       const credentialsFound = await this.userRepository.findCredentials(
+         foundUser.id,
+       );
+       if (!credentialsFound) {
+         throw new HttpErrors.Unauthorized(invalidCredentialsError);
+       }
+
+       const passwordMatched = await compare(
+         credentials.password,
+         credentialsFound.password,
+       );
+
+       if (!passwordMatched) {
+         throw new HttpErrors.Unauthorized(invalidCredentialsError);
+       }
+
+       return foundUser;
+     }
+
+     // User --> MyUser
+     convertToUserProfile(user: MyUser): UserProfile {
+       return {
+         [securityId]: user.id.toString(),
+         name: user.username,
+         id: user.id,
+         email: user.email,
+       };
+     }
+   }
+   ```
+
+   </p>
+   </details>
+
+3. Bind `MyUserRepository` (and `MyUserCredentialsRepository` if you create your
+   own as well) to the corresponding key in your `application.ts`:
+
+   ```ts
+   import {CustomUserService} from './services/custom-user-service';
+   import {MyUserRepository, MyUserCredentialsRepository} from './repositories';
+   import {UserServiceBindings} from '@loopback/extension-authentication-jwt';
+
+   export class TestApplication extends BootMixin(
+     ServiceMixin(RepositoryMixin(RestApplication)),
+   ) {
+     constructor(options: ApplicationConfig = {}) {
+       super(options);
+       // ...other setup
+       this.component(JWTAuthenticationComponent);
+       // Bind datasource
+       this.dataSource(DbDataSource, UserServiceBindings.DATASOURCE_NAME);
+       // Bind user service
+       this.bind(UserServiceBindings.USER_SERVICE).toClass(CustomUserService),
+       // Bind user and credentials repository
+       this.bind(UserServiceBindings.USER_REPOSITORY).toClass(
+         UserRepository,
+       ),
+       this.bind(UserServiceBindings.USER_CREDENTIALS_REPOSITORY).toClass(
```

Comment:
Add a code comment that the value of `TokenServiceBindings` can be changed too if required. This is something many will definitely want to change.

Reply:
@hacksparrow I added a new section called "customization" and it has "customizing model" as the sub section, see https://github.com/strongloop/loopback-next/pull/5046/files#diff-b7edcb305a3224321ad8aea0c9573fb8R181